### PR TITLE
Working openstack prototype for switching to Ansible inventory plugins

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect #171 

This is implementing the steps described here:

https://github.com/AlanCoding/Ansible-inventory-file-examples/tree/master/plugins

inside of our task system. This is now importing openstack data in the same way that it used to.

### CODE REVIEW

**Comment by AlanCoding** [comment]
for later reference

https://github.com/ansible/ansible/blob/devel/lib/ansible/config/base.yml#L1249-L1256

File: awx/main/tasks.py
Comment (ryanpetrello) [inline awx/main/tasks.py:1947]: I don't think we had this sort of validation before now - does the API actually allow you to get into this scenario?  Also, is this preferable to letting the plugin fail in its usual way if you don't have a credential assigned?

File: awx/main/tasks.py
Comment (ryanpetrello) [inline awx/main/tasks.py:1954]: Given `if src in Inventory.SOURCE_PLUGINS`, is this `else` block of code really necessary?

File: awx/main/tasks.py
Comment (ryanpetrello) [inline awx/main/tasks.py:1945]: This isn't a requirement for this PR, but just something on my mind as we continue to work on this script -> plugin migration in general.

I've always really disliked the way that our inventory script building is a giant chain of `if src == 'xyz'` checks in `tasks.py` here.  It would be _really_ nice to take this opportunity to move this config building logic into separate, testable code elsewhere, maybe something like `InventoryUpdate.build_dynamic_inventory_file(**kwargs)`, which would write a file and return a path for `-i` ?

**Review by ryanpetrello — APPROVED** [review]
This change looks good to me in general, and matches what I'd expect reading the ansible core docs/feature description.  To avoid introducing issues for awx users, I think I'd prefer to hold off on merging this until more plugins upstream are implemented and have some time to bake (maybe until around the time @AlanCoding returns?)

File: awx/main/tasks.py
Comment (AlanCoding) [inline awx/main/tasks.py:1941]: This is a major sore point here. From discussions with the Ansible core team, we don't expect that all inventory scripts will be converted to plugins before the 2.5 release. Because of that, I think that the `Inventory.SOURCE_PLUGINS` dictionary will need to contain data on which version of Ansible _that particular plugin_ was introduced in. I think I remember @nitzmahone making this point.

Since some older versions of Ansible may not have some particular plugin, we'll have to continue to keep the inventory scripts until each ages out of the versions of Ansible that we support.

**Comment by AlanCoding** [comment]
This is punted for a later time, dependent on when Ansible core gets these in.

File: awx/main/tasks.py
Comment (AlanCoding) [inline awx/main/tasks.py:759]: I still want to salvage this for the current development cycle.

Comment (AlanCoding) [reply awx/main/tasks.py:759]: link #864

**Pull request closed without merging**
